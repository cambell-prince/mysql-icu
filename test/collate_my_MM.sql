-- Test ICU charset collation
set names utf8;

drop database if exists collate_my_MM;
create database collate_my_MM;
use collate_my_MM;
create table testCollate (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    word VARCHAR(128)
) collate utf8_icu_my_MM_ci;
insert into testCollate (word) values ('ခါ'),('ကို'),('ကာ'),('ကူ'),('ကီ'),('ကု'),('ခို'),('ခ'),('ကြ'),('ကြွ'),('ချ'),('ခှ'),('ကျ'),('ကှ'),('ကံ'),('ကက်'),('ခေါ်');
-- select * from testCollate order by word into outfile '/home/keith/projects/mysql-icu/mysql-icu/test/collate_my_MM.result1';
-- select * from testCollate order by word collate utf8_icu_custom_ci into outfile '/home/keith/projects/mysql-icu/mysql-icu/test/collate_my_MM.result2';

create table ordered_my_MM (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    word VARCHAR(128)
) collate utf8_icu_my_MM_ci;
insert into ordered_my_MM (word) select word from testCollate order by word;

create table ordered_custom (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    word VARCHAR(128)
) collate utf8_icu_my_MM_ci;
insert into ordered_custom (word) select word from testCollate order by word collate utf8_icu_custom_ci;

-- This shouldn't output anything if the custom is using the same rules as my_MM
select ordered_my_MM.id, ordered_custom.id, ordered_my_MM.word  from ordered_my_MM left join ordered_custom on (ordered_my_MM.word = ordered_custom.word) where ordered_my_MM.id != ordered_custom.id;

create table ordered_uca (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    word VARCHAR(128)
) collate utf8_icu_my_MM_ci;
insert into ordered_uca (word) select word from testCollate order by word collate utf8_unicode_ci;

-- This will output the differences between Myanmar and Unicode collation
select ordered_my_MM.id as my_MM, ordered_uca.id as uca, ordered_my_MM.word  from ordered_my_MM left join ordered_uca on (ordered_my_MM.word = ordered_uca.word) where ordered_my_MM.id != ordered_uca.id;

-- Now test using UTF16 charset
create table testCollate16 (
    id INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
    word VARCHAR(128)
) collate ucs2_icu_custom_ci;
insert into testCollate16 (word) values ('ခါ'),('ကို'),('ကာ'),('ကူ'),('ကီ'),('ကု'),('ခို'),('ခ'),('ကြ'),('ကြွ'),('ချ'),('ခှ'),('ကျ'),('ကှ'),('ကံ'),('ကက်'),('ခေါ်');
select word from testCollate16 order by word collate ucs2_icu_custom_ci;

