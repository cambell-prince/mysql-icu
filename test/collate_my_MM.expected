my_MM	uca	word
5	1	ကံ
7	2	ကက်
1	3	ကာ
6	4	ကို
2	5	ကီ
3	6	ကု
4	7	ကူ
11	10	ကြွ
10	11	ကှ
15	13	ခို
13	15	ခါ
